src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/haflows/HaFlowHistorySpec.groovy

```diff
@@ -0,0 +1,258 @@
+package org.openkilda.functionaltests.spec.flows.haflows
+
+import static org.openkilda.functionaltests.extension.tags.Tag.LOW_PRIORITY
+import static org.openkilda.testing.Constants.WAIT_OFFSET
+import static org.openkilda.testing.service.floodlight.model.FloodlightConnectMode.RW
+
+import org.openkilda.functionaltests.HealthCheckSpecification
+import org.openkilda.functionaltests.error.HistoryMaxCountExpectedError
+import org.openkilda.functionaltests.extension.failfast.Tidy
+import org.openkilda.functionaltests.extension.tags.Tags
+import org.openkilda.functionaltests.helpers.Wrappers
+import org.openkilda.functionaltests.helpers.model.HaFlowExtended
+import org.openkilda.messaging.info.event.IslChangeType
+import org.openkilda.messaging.payload.flow.FlowState
+import org.openkilda.messaging.payload.history.HaSubFlowPayload
+import org.openkilda.model.FlowEncapsulationType
+import org.openkilda.model.SwitchId
+import org.openkilda.model.history.DumpType
+import org.openkilda.northbound.dto.v2.flows.BaseFlowEndpointV2
+import org.openkilda.northbound.dto.v2.haflows.HaFlowPatchPayload
+import org.openkilda.testing.service.northbound.model.HaFlowActionType
+
+import org.springframework.web.client.HttpClientErrorException
+import spock.lang.Narrative
+
+@Narrative("""Verify that history records are created for the basic actions applied to Ha-Flow.""")
+class HaFlowHistorySpec extends HealthCheckSpecification {
```

Comment:
Please, leave one or two tests in this class normal priority while decreasing the priority of others to 'Low'.

Reply:
Done.